gammapy/background/energy_offset_array.py

```diff
@@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
         ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
                                                      [self.energy.value,
                                                       self.offset.value])
-        return ev_cube_hist
+        energy_bin = np.diff(ev_cube_edges[0])
+        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
+        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
+        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
```

Comment:
ok!
But for the moment the read and write method I wrote are very simple, Maybe I have to take into account that the Energyoffset Array can be counts, livetime or background no?
Le 9 févr. 2016 à 19:31, Christoph Deil <notifications@github.com> a écrit :

> In gammapy/background/energy_offset_array.py:
> 
> > @@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
> >          ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
> >                                                       [self.energy.value,
> >                                                        self.offset.value])
> > -        return ev_cube_hist
> > +        energy_bin = np.diff(ev_cube_edges[0])
> > +        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
> > +        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
> > +        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
> I see you've already opened a new pull request with the EnergyOffsetBackgroundModel class: #444
> 
> Let's try to get this one finished quickly, so that you can use the functionality here in the other one.
> 
> The main thing to do here is add a test and remove the code that does the rate calculation
> (you can look at the diff on Github to check).
> 
> —
> Reply to this email directly or view it on GitHub.
>